### PR TITLE
All functions are added

### PR DESCRIPTION
**Pull request by VladimirNalivaiko**

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+  this.length++;
+
+  return node;
+}
+
+List.prototype.at = function(position){
+  var node = this.head, count = 0;
+
+  while(count < position){
```
Comment (paveldz) [inline js/main.js]: Why do you not use for cycle?

File: js/main.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  }
+
+  return node;
+}
+
+List.prototype.insertAt = function(position, data){
+  var node = this.head, count = 0;
+  var newNode = new Node(data);
+
+  while(count < position - 1){
```
Comment (paveldz) [inline js/main.js]: Why do you not use for cycle?

File: js/main.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    this.head = node;
+    this.tail = node;
+  }
+
+  this.length++;
+
+  return node;
+}
+
+List.prototype.at = function(position){
```
Comment (paveldz) [inline js/main.js]: What will happen if the position is incorrect value? (for example, nonexistent)

File: js/main.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+      this.head.previous = null; 
+    } 
+    else{
+      this.tail = null;
+    }
+  }else if(position === this.length - 1){
+    this.tail = this.tail.previous;
+    this.tail.next = null;    
+  }else{
+    while(count < position){
```
Comment (paveldz) [inline js/main.js]: use 'at' method to find node for deleting

File: js/main.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+  while(count < position){
+    node = node.next;
+    count++;
+  }
+
+  return node;
+}
+
+List.prototype.insertAt = function(position, data){
```
Comment (paveldz) [inline js/main.js]: Use 'at' here method if it possible

**Review by paveldz — CHANGES_REQUESTED** [review]

File: js/main.js (hunk 3 above)
Comment (VladimirNalivaiko) [reply js/main.js]: okey, i will handle exeption

File: js/main.js [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+  var node = this.head, count = 0;
+  var newNode = new Node(data);
+  try{
+    this.checkRange(position);
```
Comment (paveldz) [inline js/main.js]: checkRange function is called twice (the second time inside 'at' function)

File: js/main.js [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+  }
+
+  this.length++;
+
+  return node;
+}
+
+List.prototype.checkRange = function(position){
+  if(position > this.length || position < 0){
+    throw new RangeError("Uncorrect position in 'at' function");
```
Comment (paveldz) [inline js/main.js:61]: Not only 'at' function can throw this exception. I think the error message is incorrect

File: js/main.js [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+  }
+}
+
+List.prototype.at = function(position){
+  var node = this.head;
+  try{
```
Comment (paveldz) [inline js/main.js]: Catching of exception here - is not good idea. Consumer of your code should be responsible for this. Your code should just throw exceptions

**Review by paveldz — CHANGES_REQUESTED** [review]

**Review by paveldz — APPROVED** [review]

**Pull request merged**